### ISSUE TITLE
[Feature Request] - Mirage side modification for vllm integration

### ISSUE THREAD

**Issue by NorthmanPKU** [mirage-project/mirage#522]
### Feature Description

Mirage side TODOs (We should finish these in a new `vllm_integration` branch): 
- [x] Expose a function above `PersistentKernel` in mpk to accept metadatas (and mirage friendly graph) from vllm side and build the final graph, compile, and return callable result.
- [x] Make it controllable to only compile inference process so that we can let scheduling be handled outside.
- [x] Build the `python/mirage/models/` directory and pack Qwen3 graph as a `Qwen3GraphBuilder` to automatically build the graph depending on the case. (Reference: `demo/qwen3/demo.py`)
- [x] We should organize our `python/mirage` better. At least put mpk related files in a `mpk/` directory. Influence to other places should also be considered.
- [ ] As a next step: Build real transformation logic to get final graph.

vllm side related issue: https://github.com/vllm-project/vllm/issues/22201

----------------------------------------------------------------------------------------

### What do we need and where are they defined, maintained, get updated and how can we get them?
- ⭕️step [total_num_requests, ]
  - We don't need to maintain this. But it is still passed to mirage as `positions` tensor of `*args`.
- ❌tokens [total_num_requests, max_seq_length]
  - We don't need to maintain this.
- ✅input_tokens [max_num_batched_tokens, 1]
  - This is allocated as `input_ids` inside `GPUModelRunner` as `CpuGpuBuffer`.
  - This is updated every time before a forward pass in `GPUModelRunner`'s `_prepare_inputs()`->`_prepare_input_ids()`.
  - We'll get it directly from `*args` as the second parameter of our callable when it is called for the first time.
- ❌output_tokens [max_num_batched_tokens, 1]
  - We should not write to output_tokens. Instead, we should give the final logits back to vllm and let vllm do the sampling work.
- ❌num_new_tokens [total_num_requests, ]
  - We don't need to maintain this.
- ❌prompt_lengths [total_num_requests, ]
  - We don't need to maintain this.
- ✅qo_indptr_buffer [max_num_batched_requests + 1]
  - This is allocated as `query_start_loc` inside `GPUModelRunner` as `CpuGpuBuffer`.
  - This is updated every time before a forward pass in `GPUModelRunner`'s `_prepare_inputs()`.
  - We'll get it from common_attn_metadata in `MirageAttentionMetaDataBuilder`, register the tensor into `MirageAttentionMeta`, and pass it to our callable in the first call.
- ✅paged_kv_indptr_buffer [max_num_batched_requests + 1]
  - This is allocated as `paged_kv_indptr` in `MirageAttentionMetadataBuilder`.
  - This is updated every time before a forward pass in `MirageAttentionMetadataBuilder.build()` by calculating cumsum based on `common_attn_metadata.seq_lens_cpu`.
  - We'll get it from attn_metadata during the first call of our callable.
- ✅paged_kv_indices_buffer [max_num_pages]
  - This is allocated as `paged_kv_indices` in `MirageAttentionMetadataBuilder`.
  - This is updated every time before a forward pass in `MirageAttentionMetadataBuilder.build()` based on `common_attn_metadata.block_table_tensor` and `paged_kv_indptr`.
  - We'll get it from attn_metadata during the first call of our callable.
- ✅paged_kv_last_page_len_buffer [max_num_batched_requests]
  - This is allocated as `paged_kv_last_page_len` in `MirageAttentionMetadataBuilder`.
  - This is updated every time before a forward pass in `MirageAttentionMetadataBuilder.build()` by calculating based on `common_attn_metadata.seq_lens_cpu` and `paged_size`.
  - We'll get it from attn_metadata during the first call of our callable.
- ✅k/v_cache [num_layers, max_num_pages, page_size, num_kv_heads // world_size, head_dim]
  - In vllm k/v_caches are allocated as separate tensors for each layer in GPUModelRunner.
  - k/v_caches are bound to `GPUModelRunner`'s kvcaches and `compilation_config.static_forward_context`, which is an dict from layer name to Attention class. We can get kvcaches from this class as `compilation_config.static_forward_context[layer_name].kvcache[0]`.
- ✅model weights
  - Model weights are loaded and maintained by vllm side models.
  - We'll get them directly from `*args` as parameters of our callable when it is called for the first time. We will get weight names and transfer them to our name formats when `MirageBackend` is called.
